File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.cpp
class Solution {
public:
    bool isVowel(char s) {
        return s == 'a' || s == 'e' || s == 'i' || s == 'o' || s == 'u' ||
                s == 'A' || s == 'E' || s == 'I' || s == 'O' || s == 'U';
    }

    string sortVowels(string s) {
        vector<char> vowels;

        for(int i=0;i<s.size();i++) {
            if(isVowel(s[i])) {
                vowels.push_back(s[i]);
            }
        }

        sort(vowels.begin(), vowels.end());
        int ind = 0;

        for(int i=0;i<s.size();i++) {
            if(isVowel(s[i])) {
                s[i] = vowels[ind];
                ind++;
            }
        }

        return s;
    }
};

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.java
class Solution {
    private boolean isVowel(char s) {
        return s == 'a' || s == 'e' || s == 'i' || s == 'o' || s == 'u' ||
               s == 'A' || s == 'E' || s == 'I' || s == 'O' || s == 'U';
    }

    public String sortVowels(String s) {
        int n = s.length();
        ArrayList<Character> vowels = new ArrayList<>();

        for(int i = 0; i < n; i++) {
            if(isVowel(s.charAt(i))) {
                vowels.add(s.charAt(i));
            }
        }

        Collections.sort(vowels);

        int ind = 0;
        StringBuilder str = new StringBuilder(s);

        for(int i = 0; i < n; i++) {
            if(isVowel(str.charAt(i))) {
                str.setCharAt(i, vowels.get(ind++));
            }
        }

        return str.toString();
    }
}